security/cmd/generate_cert/main.go

```diff
@@ -47,37 +47,32 @@ var (
 	keySize        = flag.Int("key-size", 2048, "Size of the generated private key")
 )
 
-func fatalf(template string, args ...interface{}) {
-	log.Errorf(template, args)
-	os.Exit(-1)
-}
-
 func checkCmdLine() {
 	flag.Parse()
 
 	hasCert, hasPriv := len(*signerCertFile) != 0, len(*signerPrivFile) != 0
 	if *isSelfSigned {
 		if hasCert || hasPriv {
-			fatalf("--self-signed is incompatible with --signer-cert or --signer-priv.")
+			log.Fatalf("--self-signed is incompatible with --signer-cert or --signer-priv.")
```

Comment:
We don't want command-line tools to be using logging for their primary output to the user. It produces a terrible interactive experience.

Reply:
The original implementation of the “fatalf” is print it to the log. I think you are right, so I modified implementation of the “fatalf” in generate_cert/main.go and generate_csr/main.go.